beesdoo_shift/wizard/validate_attendance_sheet.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+from openerp import _, api, exceptions, fields, models
+from openerp.exceptions import UserError, ValidationError
+
+
+class ValidateAttendanceSheet(models.TransientModel):
+    _name = "beesdoo.shift.sheet.validate"
+    _description = """Check the user name and validate sheet.
+    Useless for users in group_cooperative_admin"""
+    _inherit = ["barcodes.barcode_events_mixin"]
+
+    @api.multi
+    def _get_active_sheet(self):
+        sheet_id = self._context.get("active_id")
+        sheet_model = self._context.get("active_model")
```

Comment:
Why didn't you set `active_sheet` as a field. Access would be simpler to write and would avoid the browse for each access to its field (even if most probably cached).

Reply:
It's true ! Changes are done, I don't know why I used this painful way.